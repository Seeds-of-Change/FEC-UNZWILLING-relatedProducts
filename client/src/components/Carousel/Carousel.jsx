import React from 'react';
// import PropTypes from 'prop-types';
import style from './styles.css';
import ProductCard from '../ProductCard/ProductCard';

export default function Carousel(props) {
  const {
    products,
  } = props;

  return (
    <div className={style.wrapper}>

      {products.map((product, index) => (
        <ProductCard product={product} />
      ))}

    </div>
  );
}

// Carousel.defaultProps = {
//   header: null,
//   features: null,
//   updateActive: null,
//   activeIndex: 0,
// };

// Carousel.propTypes = {
//   features: PropTypes.arrayOf(
//     PropTypes.shape({
//       heading: PropTypes.string,
//       description: PropTypes.string,
//       posX: PropTypes.number,
//       posY: PropTypes.number,
//     }),
//   ),
//   header: PropTypes.string,
//   updateActive: PropTypes.func,
//   activeIndex: PropTypes.number,
// };
