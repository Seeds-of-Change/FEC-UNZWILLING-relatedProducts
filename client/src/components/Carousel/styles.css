.wrapper {
  background: #F2F0EE;
  position: relative;
}

.cards {
  margin: 100px auto;
  max-width: 80%;
  overflow-x: hidden;
  overflow-y: hidden;
  white-space: nowrap;
}

.cards.active {
  display: block;
}

button.arrows {
  position: absolute;
  border: none;
  padding: 0;
  outline:inherit;
  width: 50px;
  height: 50px;
  line-height: 50px;
  top: 40%;
  cursor: pointer;
}

.arrows:focus {
  outline: 0;
}

.arrows:hover {
  background: white;
  border-color: white;
  color: #111;
  opacity: .5;
}

.arrows_left {
  left: 5%;
}

.arrows span{
  display:block;
  height:25px;
  width:25px;
}
.arrows_left > span {
  border-bottom: 2px solid;
  border-left: 2px solid;
  transform: rotateZ(45deg);
}

.arrows_right {
  right: 5%;
}
.arrows_right > span {
  border-top: 2px solid;
  border-right: 2px solid;
  transform: rotateZ(45deg);
}

.arrows:after {
  position: absolute;
  display: inline-block;
  content: "";
  width: 25px;
  height: 25px;
  top: 50%;
  left: 50%;
}
