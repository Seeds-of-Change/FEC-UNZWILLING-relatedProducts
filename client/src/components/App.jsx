import React from 'react';
import axios from 'axios';
import style from './styles.css';
import Carousel from './Carousel/Carousel';

export default class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      products: [],
      currentProductId: 1,
    };
    this.getRelatedProducts = this.getRelatedProducts.bind(this);
    this.updateProduct = this.updateProduct.bind(this);
  }

  componentDidMount() {
    const { currentProductId } = this.state;
    this.getRelatedProducts(currentProductId);
  }

  getProduct(id) {
    this.setState({ currentProductId: id });
    this.getRelatedProducts(id);
  }

  getRelatedProducts(id) {
    axios.get(`http://localhost:3001/relatedProducts/${id}`)
      .then((response) => {
        this.setState({
          products: response.data,
        });
      })
      .catch((error) => {
        console.log('ERROR: ', error);
      });
  }

  updateProduct(updatedProduct) {
    const { currentProductId } = this.state;
    axios.post(`http://localhost:3001/products/${updatedProduct.id}`, updatedProduct)
      .then((response) => {
        this.getRelatedProducts(currentProductId);
      })
      .catch((error) => {
        console.log('ERROR: ', error);
      });
  }

  render() {
    const { products } = this.state;
    return (
      <div className={style.relatedProducts}>
        <h1 className={style.header}>You may also like</h1>
        <Carousel products={products} updateProduct={this.updateProduct} />
      </div>
    );
  }
}
