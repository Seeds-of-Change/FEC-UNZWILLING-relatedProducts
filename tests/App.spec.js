import React from 'react';
import renderer from 'react-test-renderer';
import { mount, shallow } from 'enzyme';
import App from '../client/src/components/App';
import Carousel from '../client/src/components/Carousel/Carousel';
// import ProductCard from '../client/src/components/ProductCard/ProductCard';

describe('App', () => {
  test('snapshot renders', () => {
    const component = renderer.create(<App />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders the Features Details Component', () => {
    const wrapper = mount(<App />);
    expect(wrapper.find(Carousel).length).toEqual(1);
  });

  // test('renders the Image Viewer Component', () => {
  //   const wrapper = mount(<App />);
  //   expect(wrapper.find(ProductCard).length).toEqual(1);
  // });
});

describe('My Test Suite', () => {
  test('My Test Case', () => {
    expect(true).toEqual(true);
  });
});