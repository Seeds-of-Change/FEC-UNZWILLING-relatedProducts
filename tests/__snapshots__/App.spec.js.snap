// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot renders 1`] = `
<div
  className="App"
>
  <h1
    className="header"
  >
    You may also like
  </h1>
  <div
    className="wrapper"
  >
    <button
      className="arrows arrows_left"
      onClick={[Function]}
      type="button"
    >
      <span />
    </button>
    <div
      className="cards"
    />
    <button
      className="arrows arrows_right"
      onClick={[Function]}
      type="button"
    >
      <span />
    </button>
  </div>
</div>
`;
